### DISCUSSION TITLE
How to run xsh scripts located in the path on current dir

### DISCUSSION THREAD

**Discussion by pannet1** [xonsh/xonsh#5147] [Q&A] [answered]
is there a way i can simply call the `xsh` script without calling it as a `xonsh` argument.

**Comment by bestlem** [comment, marked as answer]
Begin the script with a shebang e.g.:
```xsh
cd /tmp

echo @("""#!/usr/bin/env xonsh
echo world
""") > hello.xsh

chmod +x hello.xsh

./hello.xsh
# world
```

**Reply by pannet1** [reply]
will it work if `hello.xsh` is not in the current working directory.

**Reply by bestlem** [reply]
> will it work if `hello.xsh` is not in the current working directory.

No that is a separate issue.

It works like any other executable. It needs to be called by a path that contains a directory or is on your $PATH

**Comment by pannet1** [comment]
thanks mate it works now. i needed to try on a new shell after doing the changes

```
 🐍(copier) $lsd.xsh
__pycache__
static
templates
tests
```

**Reply by anki-code** [reply]
Please read [xonsh cheatsheet](https://github.com/anki-code/xonsh-cheatsheet/) to solve many other questions.
